### PR TITLE
Readline and README

### PR DESCRIPTION
**Pull request by Dymirt**
This commit is adding deadline library to the project and Update README file with project requirements.
Features:
- conditionally changes readline path in makefile based on OS

### CODE REVIEW

**Comment by Dymirt** [comment]
Hi! Waiting for review.

**Comment by Dymirt** [comment]
There is no need to copy all data from env using malloc.

File: README.md
Comment (hipoliciak) [inline README.md:3]: Great idea to add a to-do list :)

File: srcs/driver/readline_wraper.c
Comment (hipoliciak) [inline srcs/driver/readline_wraper.c:20]: Why are you casting NULL to char* here? From what I checked, the C standard guarantees that NULL can be assigned to any pointer type without an explicit cast.

File: srcs/driver/shell.c
Comment (hipoliciak) [inline srcs/driver/shell.c:63]: Same comment as above.

**Review by hipoliciak — APPROVED** [review]
Except those two comments about casting NULL looks good. I'll see how it works after merging.

File: README.md
Comment (Dymirt) [reply README.md:3]: Thnx

**Comment by Dymirt** [comment]
Thank you for reviving PR. You right, casting NULL is redundant but not incorrect. Il fix it in next PR.

**Pull request merged**
